### PR TITLE
API: Setting marks to nil, and group_ids_by_name function.

### PR DESCRIPTION
**Pull request by hootiehoo**
- When updating marks, can now set a mark to nil (unmarked). 

Example: curl --header "Authorization: MarkUsAuth YourAuthKey" -X PUT \
    --data "Lorem%20Ipsum=nil" \
    "http://example.com/api/assignments/1/groups/5/update_marks"

- Added api function 'group_ids_by_name' that returns json/xml consisting of key:value pairs 
  of group_name:group_id for all groups associated with a given assignment.

Example: curl -H "Authorization: MarkUsAuth YourAuthKey" "http://example.com/api/assignments/1/groups/group_ids_by_name"
will return
\<?xml version="1.0" encoding="UTF-8"?>
\<groups>
  \<group_name1>31\</group_name1>
  \<group_name2>32\</group_name2>
  ...
\</groups>

### CODE REVIEW

File: app/controllers/api/groups_controller.rb [hunk 1]
```diff
@@ -142,11 +142,34 @@ def update_marks
 
     def set_mark_by_criteria(criteria, mark_to_change)
```
Comment (hootiehoo) [inline app/controllers/api/groups_controller.rb]: I don't like my use of branching if statements here, but I can't see a simpler way.

Comment (david-yz-liu) [inline app/controllers/api/groups_controller.rb]: It's okay, it's really an instance of the flexible/rubric criterion inheritance not being implemented in a nice way.

File: app/controllers/api/groups_controller.rb [hunk 2]
```diff
@@ -142,11 +142,37 @@ def update_marks
...
+        else
+          mark_to_change.mark = params[criteria.rubric_criterion_name]
+        end
       end
       mark_to_change.save
     end
+
+    # Return key:value pairs of group_name:group_id
+    def group_ids_by_name
+      reversed = Assignment.find_by_id(params[:assignment_id])
```
Comment (david-yz-liu) [inline app/controllers/api/groups_controller.rb]: Just use `find`

File: app/controllers/api/groups_controller.rb [hunk 3]
```diff
@@ -142,11 +142,37 @@ def update_marks
...
+                           .groups
+                           .inject({}) do |dict, group|
```
Comment (david-yz-liu) [inline app/controllers/api/groups_controller.rb]: This is okay (fold!) but I'd rather you use the `Hash` constructor, which takes a nested list `[ [key, val], ... ]` and returns a Hash.

**Comment by david-yz-liu** [comment]
Great, thanks @hootiehoo!

**Pull request merged**